### PR TITLE
Gallery: Display error messages for drag/paste and failed file upload

### PR DESCRIPTION
**Pull request by latter-bolden**
OTT

Fixes LAND-799

#### Error from main view:
Screenshot 2023-08-28 at 4 33 03 PM

#### Error from add curio modal:
Screenshot 2023-08-28 at 4 31 25 PM

### CODE REVIEW

File: ui/src/heap/AddCurioModal.tsx [hunk 1]
```diff
@@ -45,39 +48,42 @@ export default function AddCurioModal({
...
-      clearDraggedFile();
-    }
-    if (pastedFile) {
-      setPastedFile(null);
-    }
-  }, [clearDraggedFile, draggedFile, pastedFile]);
+    clearDragState();
+    setPastedFile(null);
+    setErrorMessage('');
+  }, [clearDragState]);
```
Comment (arthyn) [inline ui/src/heap/AddCurioModal.tsx:61]: Just calling out not sure if worth doing now though, probably could combine these into a richer object to make these state transitions cleaner.

**Review by arthyn — APPROVED** [review]
very nice! one suggestion and a callout, but lgtm

File: ui/src/heap/CurioPreview.tsx [hunk 2]
```diff
@@ -13,6 +13,7 @@ export function canPreview(text: string) {
 }
 export default function CurioPreview({ url }: { url: string }) {
   const [ready, setReady] = useState(false);
+  const [spin, setSpin] = useState(false);
```
[the remark this answers is not in the document]
Comment (latter-bolden) [reply ui/src/heap/CurioPreview.tsx]: Resolved. This was lazy, good eyes.

**Review by arthyn — APPROVED** [review]
lgtm

File: ui/src/heap/AddCurioModal.tsx (hunk 1 above)
Comment (latter-bolden) [reply ui/src/heap/AddCurioModal.tsx:61]: Fair, state balooned a bit. I don't think this should block merge, but I'll combine it when I have a chance.

**Pull request merged**